modules/@apostrophecms/i18n/index.js

```diff
@@ -97,6 +102,9 @@ module.exports = {
           locale = req.locale;
           mode = req.mode;
         }
+        if ((req.mode === 'draft') && (!self.apos.permission.can(req, 'view-draft'))) {
```

Comment:
This doesn't seem to be the purpose of this method.

Reply:
We must stop the user from entering draft mode wherever it is possible for them to do that if they do not have permission to do that. This method is one of the two places where `req.mode` can be set. So this is a good place for this check, otherwise it is potentially needed in hundreds of places.